File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        cnt = 0
        l = 0
        r = len(nums) - 1
        while l < r:
            temp = nums[l] + nums[r]
            if temp == k:
                cnt += 1
                l += 1
                r -= 1
            elif temp < k:
                l += 1
            else:
                r -= 1
        return cnt

File: 1798-max-number-of-k-sum-pairs/max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        int left = 0, right = nums.length - 1;
        Arrays.sort(nums);
        int cnt = 0;
        while(left < right){
            int sum = nums[left] + nums[right];
            if(sum == k){
                left++;
                cnt++;
                right--;
            }
            else if(sum < k){
                left++;
            }
            else{
                right--;
            }
        }
        return cnt;
    }
    
}